Content.Shared/Movement/Pulling/Systems/PullingSystem.cs

```diff
@@ -275,7 +428,56 @@ private void AddPullVerbs(EntityUid uid, PullableComponent component, GetVerbsEv
 
     private void OnRefreshMovespeed(EntityUid uid, PullerComponent component, RefreshMovementSpeedModifiersEvent args)
     {
-        args.ModifySpeed(component.WalkSpeedModifier, component.SprintSpeedModifier);
+        if (TryComp<HeldSpeedModifierComponent>(component.Pulling, out var itemHeldSpeed) && component.Pulling.HasValue)
+        {
+            var (walkMod, sprintMod) =
+                _clothingMoveSpeed.GetHeldMovementSpeedModifiers(component.Pulling.Value, itemHeldSpeed);
+            args.ModifySpeed(walkMod, sprintMod);
+        }
+
+        if (TryComp<HeldSpeedModifierComponent>(component.Pulling, out var heldMoveSpeed) && component.Pulling.HasValue)
+        {
+            var (walkMod, sprintMod) = (args.WalkSpeedModifier, args.SprintSpeedModifier);
+
+            switch (component.GrabStage)
+            {
+                case GrabStage.No:
+                    args.ModifySpeed(walkMod, sprintMod);
+                    break;
+                case GrabStage.Soft:
+                    args.ModifySpeed(walkMod * 0.9f, sprintMod * 0.9f);
+                    break;
+                case GrabStage.Hard:
+                    args.ModifySpeed(walkMod * 0.7f, sprintMod * 0.7f);
+                    break;
+                case GrabStage.Suffocate:
+                    args.ModifySpeed(walkMod * 0.4f, sprintMod * 0.4f);
+                    break;
+                default:
+                    args.ModifySpeed(walkMod, sprintMod);
+                    break;
+            }
+            return;
+        }
+
+        switch (component.GrabStage)
+        {
+            case GrabStage.No:
+                args.ModifySpeed(component.WalkSpeedModifier, component.SprintSpeedModifier);
+                break;
+            case GrabStage.Soft:
+                args.ModifySpeed(component.WalkSpeedModifier * 0.9f, component.SprintSpeedModifier * 0.9f);
```

Comment:
```suggestion
                args.ModifySpeed(component.WalkSpeedModifier * 0.9f, component.SprintSpeedModifier * 0.9f);
```
Unhardcode these as well.